lib/modem_info/modem_info_json.c

```diff
@@ -161,23 +169,23 @@ static int device_data_add(struct device_param *device, cJSON *json_obj)
 	total_len = json_add_data(&device->modem_fw, json_obj);
 	total_len += json_add_data(&device->battery, json_obj);
 	total_len += json_add_str(json_obj, "board", device->board);
-	total_len += json_add_str(json_obj, "appVer", device->app_version);
+	total_len += json_add_str(json_obj, "appVersion", device->app_version);
 	total_len += json_add_str(json_obj, "appName", device->app_name);
 
 	return total_len;
 }
 
-int modem_info_json_string_encode(struct modem_param_info *modem, char *buf)
+int modem_info_json_object_encode(struct modem_param_info *modem,
+				  cJSON *root_obj)
 {
 	int total_len = 0;
 	int ret;
 
-	cJSON *data_obj		= cJSON_CreateObject();
 	cJSON *network_obj	= cJSON_CreateObject();
```

Comment:
We now extend the root object with `json_add_str` and `cJSON_PrintUnformatted(network_obj)` instead of `json_add_obj`. Shouldn't we release this and other objects allocated in this function on our own then?

Reply:
Of course, good catch.